types/node/compatibility/disposable.d.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * Polyfills for the explicit resource management types added in TypeScript 5.2.
+ * TODO: remove once this package no longer supports TS 5.1, and replace with a
+ * <reference> to TypeScript's disposable library in index.d.ts.
+ */
+
+interface SymbolConstructor {
```

Comment:
I get your point, although `RelativeIndexable` _is_ merged forward into the global-scope `Array` etc. – it's not an interface used within @types/node, it exists solely to play lib.es2022.array and friends. Those definitions just need to go, and it's up for debate whether that needs to wait on a new version or not, given that the compiler will report back to the user that they're missing lib.es2022.\_ if they're using the `.at()` methods without the correct TS libs.

Regarding `Disposable`, there are probably specific barriers to sequestering this away. Defining some sort of package-local `NodeDisposable` would still necessitate monkeypatching `@@dispose` and `@@asyncDispose` onto the global `SymbolContructor`; I don't think that it's possible to define the interfaces without touching the global scope in some way or another? The merge-forward technique doesn't appear to be an option, since something like
```ts
interface Disposable {}
interface NodeDisposable extends Disposable {}
```
would make the checker consider every symbol "disposable" unless lib.esnext.disposable is also loaded.

Reply:
We reintroduced `typesVersions` to support `Buffer` and `Uint8Array` now being generic, so you may be able to leverage `typesVersions` to load TS-specific types.